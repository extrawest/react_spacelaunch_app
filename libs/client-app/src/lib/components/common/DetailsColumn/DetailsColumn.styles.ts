import type { SxProps } from '@mui/material';

const wrapper: SxProps = {
  alignItems: 'center',
  textAlign: 'center',
};

const detailValue: SxProps = {
  fontWeight: 700,
};

export const styles = {
  wrapper,
  detailValue,
};
