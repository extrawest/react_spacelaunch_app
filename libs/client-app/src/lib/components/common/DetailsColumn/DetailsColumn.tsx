import type { FC } from 'react';

import { Stack, Typography } from '@mui/material';

import { styles } from './DetailsColumn.styles';
import type { DetailsColumnProps } from './DetailsColumn.types';

export const DetailsColumn: FC<DetailsColumnProps> = ({
  icon,
  title,
  details,
}) => {
  return (
    <Stack
      spacing={2.5}
      sx={styles.wrapper}
    >
      {icon}
      <Typography variant="subtitle1">{title}</Typography>
      <Stack spacing={1.25}>
        {details.map((detail) => (
          <div key={detail.label}>
            <Typography variant="body1">{detail.label}</Typography>
            <Typography
              variant="body1"
              sx={styles.detailValue}
            >
              {detail.value}
            </Typography>
          </div>
        ))}
      </Stack>
    </Stack>
  );
};
