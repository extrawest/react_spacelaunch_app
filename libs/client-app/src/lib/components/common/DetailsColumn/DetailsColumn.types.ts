import type { ReactNode } from 'react';

export interface DetailsColumnProps {
  icon: ReactNode;
  title: string;
  details: DetailParams[];
}

export interface DetailParams {
  label: string;
  value: string | number;
}
