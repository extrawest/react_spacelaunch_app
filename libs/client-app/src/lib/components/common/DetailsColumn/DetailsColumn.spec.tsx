import { render } from '@testing-library/react';

import { DetailsColumn } from './DetailsColumn';
import type { DetailParams } from './DetailsColumn.types';

describe('[component]: DetailsColumn', () => {
  const title = 'Title';
  const details: DetailParams[] = [
    {
      label: 'Label 1',
      value: 'Value 1',
    },
    {
      label: 'Label 2',
      value: 'Value 2',
    },
  ];

  test('Should match style without details', () => {
    const { baseElement } = render(
      <DetailsColumn
        icon={null}
        title={title}
        details={[]}
      />
    );

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  test('Should match style with details', () => {
    const { baseElement } = render(
      <DetailsColumn
        icon={null}
        title={title}
        details={details}
      />
    );

    expect(baseElement).toBeInTheDocument();
    expect(baseElement).toMatchSnapshot();
  });

  test('Map method should be used', () => {
    const mockMap = jest.spyOn(
      Array.prototype as { map: typeof Array.prototype.map },
      'map'
    );
    render(
      <DetailsColumn
        icon={null}
        title={title}
        details={details}
      />
    );

    expect(mockMap).toBeCalled();
  });
});
