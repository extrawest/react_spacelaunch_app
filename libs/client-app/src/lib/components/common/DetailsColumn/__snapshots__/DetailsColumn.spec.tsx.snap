// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[component]: DetailsColumn Should match style with details 1`] = `
<body>
  <div>
    <div
      class="css-1qo1efa-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Title
      </h6>
      <div
        class="css-k51q53-MuiStack-root"
      >
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Label 1
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Value 1
          </p>
        </div>
        <div>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
          >
            Label 2
          </p>
          <p
            class="MuiTypography-root MuiTypography-body1 css-ndaqpv-MuiTypography-root"
          >
            Value 2
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;

exports[`[component]: DetailsColumn Should match style without details 1`] = `
<body>
  <div>
    <div
      class="css-1qo1efa-MuiStack-root"
    >
      <h6
        class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
      >
        Title
      </h6>
      <div
        class="css-k51q53-MuiStack-root"
      />
    </div>
  </div>
</body>
`;
